### PR TITLE
click event bullet point fix

### LINKED ISSUE #35450: Click event: not any focusable element triggers a click when enter/space is pressed
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/API/Element/click_event

### What specific section or headline is this issue about?

Element: click event

### What information was incorrect, unhelpful, or incomplete?

Third bullet says:

> the Space key or Enter key is pressed while the element is focused

My reading is that any element that is currently focused will trigger a `click` when space or enter key is pressed (including elements made focusable by `tabindex`). This is not the case.

### What did you expect to see?

I'm not certain, but being focused is not the only requirement for this to work

### Do you have any supporting links, references, or citations?

Chrome bug: https://issues.chromium.org/issues/40776466 (won't fix)

Discussed in: #30263

Merged in: #30671 

### Do you have anything more you want to share?

Sample code found on Chrome bug: https://jsfiddle.net/yonran/pmkGN/

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/api/element/click_event`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/API/Element/click_event
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/api/element/click_event/index.md
* Last commit: https://github.com/mdn/content/commit/d0b23f3f26637aa405ee9ee0a0892fc6e9b742ef
* Document last modified: 2024-07-26T02:56:49.000Z

</details>

### PR DESCRIPTION
**Pull request by gagan-bhullar-tech**
### Description

PR fixes #35450

### CODE REVIEW

File: files/en-us/web/api/element/click_event/index.md [hunk 1]
```diff
@@ -12,7 +12,10 @@ An element receives a **`click`** event when any of the following occurs:
 
 - a pointing-device button (such as a mouse's primary button) is both pressed and released while the pointer is located inside the element.
 - a touch gesture is performed on the element
```
Comment (dipikabh) [inline files/en-us/web/api/element/click_event/index.md:14]:
```suggestion
- A pointing-device button (such as a mouse's primary button) is both pressed and released while the pointer is located inside the element.
- A touch gesture is performed on the element.
```

File: files/en-us/web/api/element/click_event/index.md [hunk 2]
```diff
@@ -12,7 +12,10 @@ An element receives a **`click`** event when any of the following occurs:
...
-- the <kbd>Space</kbd> key or <kbd>Enter</kbd> key is pressed while the element is focused
+- Any user interaction that is equivalent to a click, such as pressing the <kbd>Space</kbd> key or <kbd>Enter</kbd> key is pressed while the element is focused.
```
Comment (dipikabh) [inline files/en-us/web/api/element/click_event/index.md:15]:
```suggestion
- Any user interaction that is equivalent to a click occurs, such as pressing the <kbd>Space</kbd> key or <kbd>Enter</kbd> key while the element is focused.
```

File: files/en-us/web/api/element/click_event/index.md [hunk 3]
```diff
@@ -12,7 +12,10 @@ An element receives a **`click`** event when any of the following occurs:
...
+
+> [!NOTE]
+> In practice, browsers don't fire this event for non-native control elements (such as a `<div>` with `tabIndex="0"`). See <a href="https://issues.chromium.org/issues/40776466">Chromium issue</a> for the reason.
```
Comment (dipikabh) [inline files/en-us/web/api/element/click_event/index.md:18]:
```suggestion
> [!NOTE]
> In practice, browsers don't fire the `click` event for custom controls such as a `<div>` with `tabindex="0"`. To check the reason behind this behavior, see this <a href="https://issues.chromium.org/issues/40776466">Chromium issue</a>.
```

**Review by dipikabh — CHANGES_REQUESTED** [review]

File: files/en-us/web/api/element/click_event/index.md [hunk 4]
```diff
@@ -10,9 +10,12 @@ browser-compat: api.Element.click_event
...
 
-- a pointing-device button (such as a mouse's primary button) is both pressed and released while the pointer is located inside the element.
-- a touch gesture is performed on the element
-- the <kbd>Space</kbd> key or <kbd>Enter</kbd> key is pressed while the element is focused
+- A pointing-device button (such as a mouse's primary button) is both pressed and released while the pointer is located inside the element.
+- A touch gesture is performed on the element.
+- Any user interaction that is equivalent to a click occurs, such as pressing the <kbd>Space</kbd> key or <kbd>Enter</kbd> key while the element is focused.
+
+> [!NOTE]
+> In practice, browsers don't fire the `click` event for custom controls such as a `<div>` with `tabindex="0"`. To check the reason behind this behavior, see this <a href="https://issues.chromium.org/issues/40776466">Chromium issue</a>.
```
Comment (Josh-Cena) [inline files/en-us/web/api/element/click_event/index.md:18]:
```suggestion
> In practice, browsers don't fire the `click` event for custom controls such as a `<div>` with `tabindex="0"`. To check the reason behind this behavior, see this [Chromium issue](https://crbug.com/40776466).
```

Please, use Markdown

Comment (dipikabh) [reply files/en-us/web/api/element/click_event/index.md]: of course! missed this

**Review by dipikabh — APPROVED** [review]
Thank you!

**Comment by benignant** [comment]
@dipikabh, @Josh-Cena

I think the note should clearly refer only to the third bullet point, either by wording or by being inside that list item (indented 2 spaces).

The way it is now makes it sound like the `click` event does not fire at all on custom controls.

https://developer.mozilla.org/en-US/docs/Web/API/Element/click_event

**Comment by Josh-Cena** [comment]
@benignant That makes total sense. I think the note should be reworded to say "don't fire the `click` event for custom controls triggered by the third mechanism". Do you want to send a PR or issue?

**Comment by dipikabh** [comment]
Or just indenting the note should be sufficient

**Comment by Josh-Cena** [comment]
I think indenting would be too subtle. I would prefer it to be more explicit

**Comment by benignant** [comment]
I opened PR https://github.com/mdn/content/pull/36256.

**Pull request merged**